cognite/neat/rules/models/_rules/_types.py

```diff
@@ -1,6 +1,50 @@
-from typing import Annotated
+import re
+from collections.abc import Callable
+from typing import Annotated, Any
+
+from pydantic import (
+    AfterValidator,
+    BeforeValidator,
+    Field,
+    HttpUrl,
+    StringConstraints,
+    TypeAdapter,
+    ValidationInfo,
+    WrapValidator,
+)
+from rdflib import Namespace
+
+from cognite.neat.rules import exceptions
+from cognite.neat.rules.models._base import (
+    ENTITY_ID_REGEX_COMPILED,
+    VERSIONED_ENTITY_REGEX_COMPILED,
+    EntityTypes,
+    ParentClass,
+)
+from cognite.neat.rules.models.value_types import XSD_VALUE_TYPE_MAPPINGS, ValueType
+
+from .base import (
+    class_id_compliance_regex,
+    more_than_one_none_alphanumerics_regex,
+    prefix_compliance_regex,
+    property_id_compliance_regex,
+    version_compliance_regex,
+)
+
+
+def custom_error(exc_factory: Callable[[str | None, Exception], Exception]) -> Any:
```

Comment:
Suggest making all functions private, (prefix underscore), and moving them to the end of the file to make it clear that you should imports the annotated types form this module.

Reply:
I like the idea, but when moving them at the bottom it throws error:
`NameError: name '_split_parent' is not defined`